packages/core/src/internal-components/overlay/overlay.element.ts

```diff
@@ -87,8 +120,12 @@ export class CdsInternalOverlay extends CdsBaseFocusTrap {
   updated(props: Map<string, any>) {
     super.updated(props);
     const oldLayered = this.isLayered;
-    const isNested = isNestedOverlay(this.focusTrapId, this.overlayIdPrefix, FocusTrapTracker.getTrapIds());
-    const newLayered = this.focusTrap.active && isNested;
+    const newLayered = isNestedOverlay(
+      this.focusTrapId,
+      this.overlayIdPrefix,
+      FocusTrapTracker.getTrapIds(),
+      oldLayered
+    );
```

Comment:
This is the code impact of the focus trap refactor. If we make the focus trap a headless component in the future it would be helpful to migrate this into that headless component. At present, we only have one family of components that require the focus trap...